src/Octoshift/FileSystemProvider.cs

```diff
@@ -13,7 +13,7 @@ public class FileSystemProvider
 
     public virtual DirectoryInfo CreateDirectory(string path) => Directory.CreateDirectory(path);
 
-    public virtual FileStream Open(string path, FileMode mode) => File.Open(path, mode);
+    public virtual Stream Open(string path, FileMode mode) => File.Open(path, mode);
```

Comment:
Changing from `FileStream` to `Stream` in order to make more testable.

Reply:
This is not a good idea since these method are meant to reflect the underlying methods calls.